pkg/storage/couchdb/couchdbstore.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package couchdbstore
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	_ "github.com/go-kivik/couchdb" // The CouchDB driver
+	"github.com/go-kivik/kivik"
+
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+// Provider represents an CouchDB implementation of the storage.Provider interface
+type Provider struct {
+	hostURL       string
+	couchDBClient *kivik.Client
+}
+
+const blankHostErrMsg = "hostURL for new CouchDB provider can't be blank"
+
+// NewProvider instantiates Provider
+func NewProvider(hostURL string) (*Provider, error) {
+	if hostURL == "" {
+		return nil, errors.New(blankHostErrMsg)
+	}
+
+	client, err := kivik.New("couch", hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Provider{hostURL: hostURL, couchDBClient: client}, nil
+}
+
+// OpenStore opens and returns a store for the given name.
+func (p *Provider) OpenStore(name string) (storage.Store, error) {
+	exists, err := p.couchDBClient.DBExists(context.Background(), name)
+	if err != nil {
+		return nil, err
+	}
+
+	if exists {
+		db := p.couchDBClient.DB(context.Background(), name)
+
+		if dbErr := db.Err(); dbErr != nil {
+			return nil, dbErr
+		}
+
+		return &CouchDBStore{db: db}, nil
+	}
+
+	err = p.couchDBClient.CreateDB(context.Background(), name)
+	if err != nil {
+		return nil, err
+	}
+
+	db := p.couchDBClient.DB(context.Background(), name)
+
+	if err := db.Err(); err != nil {
+		return nil, err
+	}
+
+	return &CouchDBStore{db: db}, nil
+}
+
+// CloseStore closes a previously opened store.
+// CouchDB and Kivik don't really have "close" mechanisms for individual databases, so there's nothing to do here.
```

Comment:
This seems strange. Is there a way to "close" a CouchDB database?

Reply:
Is [this](https://godoc.org/github.com/go-kivik/kivik#DB.Close) what you're looking for?

Reply:
@llorllale Yep, that's it. The older version of Kivik I used initially didn't have that call, so that's why I thought it didn't exist.

I updated the method.